### PR TITLE
perl-bio-bigfile: new package

### PR DESCRIPTION
**Pull request by teaguesterling**
Adds the `perl-bio-db-bigfile` package corresponding to the `BIO::DB::Bigfile` Perl Module.

This has a handful of upstream dependency changes so it shouldn't be merged until #44413 is also merged to get kentlib up-to-date.

### CODE REVIEW

**Comment by teaguesterling** [comment]
@spackbot re-run pipeline

**Comment by teaguesterling** [comment]
@spackbot rerun pipeline

**Comment by EbiArnie** [comment]
This should probably be named perl-bio-bigfile, since Spack package names are based on the CPAN distribution name, which is Bio-BigFile.
Apart from that, builds and works fine.

File: var/spack/repos/builtin/packages/perl-bio-db-bigfile/package.py
Comment (EbiArnie) [inline var/spack/repos/builtin/packages/perl-bio-db-bigfile/package.py]: This should be named perl-bio-bigfile, the class becoming PerlBioBigfile.
Spack package names are (should be) based on the distribution name in Spack.

**Review by EbiArnie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/perl-bio-bigfile/package.py
Comment (EbiArnie) [inline var/spack/repos/builtin/packages/perl-bio-bigfile/package.py]: I think the dependencies on freetype, libpng and bzip2 are not needed here.
At least this seems to build fine without those if I also remove the linker options.

File: var/spack/repos/builtin/packages/perl-bio-bigfile/package.py
Comment (EbiArnie) [inline var/spack/repos/builtin/packages/perl-bio-bigfile/package.py]: freetype, png and bz2 are not needed here.

**Review by EbiArnie — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/perl-bio-bigfile/package.py
Comment (teaguesterling) [reply var/spack/repos/builtin/packages/perl-bio-bigfile/package.py]: These seemed very strange to me, but I couldn't get it to build without them. Let me try again without the linker options. Maybe something was particularly weird on my machine as I was testing.

File: var/spack/repos/builtin/packages/perl-bio-bigfile/package.py
Comment (teaguesterling) [reply var/spack/repos/builtin/packages/perl-bio-bigfile/package.py]: Thought it was weird that these were needed from kentutils.

Comment (teaguesterling) [reply var/spack/repos/builtin/packages/perl-bio-bigfile/package.py]: Tested and worked without these. Not sure what was going on when they were originally needed.

**Comment by ArnieSan** [comment]
This looks good to me and can be merged.
This is my other account though and I'm AFK for two weeks. Will integrate then if no one beats me to it.

**Comment by bernhardkaindl** [comment]
@ArnieSan: You requested changes as @EbiArnie: This PR is blocked until it submits an approval for this PR.

Then, we can merge (in principle).

**Comment by EbiArnie** [comment]
Builds and works fine.
Checksums are fine.

One thing that can be addressed later is a test failure for:
`spack -e myenv install --test root`.
I looked at that but could not immediately figure out what's wrong.

**Review by EbiArnie — APPROVED** [review]
OK

**Pull request merged**
